pkg/api/types.go

```diff
@@ -738,7 +771,35 @@ type Endpoints struct {
 	TypeMeta   `json:",inline"`
 	ObjectMeta `json:"metadata,omitempty"`
 
-	Endpoints []string `json:"endpoints,omitempty"`
+	Endpoints []Endpoint `json:"endpoints,omitempty"`
+}
+
+type Endpoint struct {
```

Comment:
I hear that, but I'm afraid this starts to wander away from the "simple API that isn't deeply tied to kubernetes".  That aside, Spec/Status  is only on top-level objects, but here the top-level object is Endpoints - plural.

I'll do this if you think it is important, but my instinct is to leave it until we have clear need.